File: php/spec/tokenizer/comments.spec.php
<?php declare(strict_types=1);

require_once(__DIR__ . '/util.php');

describe('Comment tokenization', function() {
  given('input', function() {
    return <<<DOC
> note
> more notes
    >    note
    >
DOC;
  });

  it('works as specified', function() {
    $instructions = inspectTokenization($this->input);

    expect($instructions)->toMatchSnapshot('spec/tokenizer/snapshots/comment.snap.json');
  });
});

File: ruby/spec/parser/comments.spec.rb
# frozen_string_literal: true

input = <<~DOC.strip
> note
> more notes
    >    note
    >
DOC

describe Enolib::Parser do
  describe 'Comments' do
    context = Enolib::Context.new(input)

    it 'parses as expected' do
      expect(context.document).to match_snapshot
    end
  end
end
